tests/e2e/users/add-user.wdio-spec.js

```diff
@@ -0,0 +1,105 @@
+const utils = require('../../utils');
+const faker = require('faker');
+const chai = require('chai');
+const usersAdminPage = require('../../page-objects/admin/user.wdio.page');
+const placeFactory = require('../../factories/cht/contacts/place');
+const loginPage = require('../../page-objects/login/login.wdio.page');
+const personFactory = require('../../factories/cht/contacts/person');
+
+const onlineUserRole = 'Program Officer';
+const offlineUserRole = 'CHW';
+const username = faker.name.firstName().toLocaleLowerCase();
+const password = 'Jacktest@123';
+const incorrectpassword = 'Passwor';
+const places = placeFactory.generateHierarchy();
+const district_hospital = places.find((place) => place.type === 'district_hospital');
+
+const person = personFactory.build(
+  {
+    parent: {
+      _id: district_hospital._id,
+      parent: district_hospital.parent
+    }
+  });
+
+const docs = [...places, person];
+
+describe('User Test Cases ->', () => {
+
+  before(async () => {
+    await utils.saveDocs(docs);
+    await loginPage.cookieLogin();
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+  });
+
+  beforeEach(async () => {
+    await usersAdminPage.goToAdminUser();
+    await usersAdminPage.openAddUserDialog();
+  });
+
+  describe('Creating Users ->', () => {
+
+    it('should add user with valid password', async () => {
+      await usersAdminPage.inputAddUserFields(username, 'Jack', onlineUserRole, district_hospital.name, 
+        person.name, password);
+      await usersAdminPage.saveUser();
+      chai.expect(await usersAdminPage.getAllUsernames()).to.contain.members([username]);
+      await utils.deleteUsers([{ username: username }]);
+    });
+  });
+
+  describe('Invalid entries -> ', () => {
```

Comment:
All of these tests follow the same pattern, which is terrific. We can clean this code up a lot by using [Mocha's dynamic test functionality](https://mochajs.org/#dynamically-generating-tests). The one outlier is the first test that does a `toBe` assertion, but I think it's fine to change that to a `toContain` to fit the pattern.

Please change this code to work similar to the above doc link. It should end up with an object containing some data, then a loop to perform the actions and assertion.

Reply:
I did think about it initially but decided against it because creating one common function to test across `password`, `username`, `place` and `contact` validations would have required a bunch of `if..else..` conditions which I wanted to avoid. However I see we can still do it for the `password` tests at-least: making that change.